### ISSUE TITLE
Training our very first model

### ISSUE THREAD

**Issue by shivam7569** [andy6975/Memoir#21]
The pitch is all set, it's time for a masterstroke!
We are gonna train an autoencoder in its simplest variation...(drum roll)...THE VANILLA AUTOENCODER...

Here are the details:

* Make a separate file for the code of model, for modularity.
* The model should be flexible enough as to depend on user's inputs, i.e., the number of layers, their respective number of neurons, the hyperparameters, everything should be taken as input.
* This will help in studying the variations of this architecture, like undercomplete, overcomplete, etc.
* Use Tensorflow 1.14
* You have to let others know on which variation you're willing to work. There are four people so I was thinking of dividing into two groups for undercomplete and overcomplete (open for suggestions).
* Everything has to be recorded while training and hyper-parameter optimization, which will be framed into a report and uploaded on github (this was the whole essence of this project).
* The more specific you are, the better.
* Visualize the activation function, each intermediate layer, variation of loss function, output at various epochs, the change in gradients, distribution of weights, graph of the model, histograms on weights, etc., for the final trained model.
* Remember to make the checkpoints at various epochs, this helps in restart the training and testing the model after training.
* Please give your inputs on how can we enhance our experience.

Hope you'd enjoy working on this!
